earth_enterprise/src/fusion/autoingest/StorageManager.h

```diff
@@ -38,9 +43,55 @@ class StorageManaged {
     StorageManaged() : timestamp(0), filesize(0) {}
 };
 
+template<class AssetType, class AssetStorageType> class AssetHandleInterface;
+
+template<class AssetType, class AssetStorageType>
+class AssetSerializerLocalXML
+{
+  public:
+    AssetSerializerLocalXML() {}
+    std::shared_ptr<AssetType> Load(const std::string &boundref)
+    {
+      return AssetType::Load(boundref);
```

Comment:
Minor: It would be nice if we could get this Load code out of the assets as well. Since AssetType::Load is autogenerated, we could change the autogeneration to make it a part of this class. To go all the way we'd need to move the Load logic inside each asset class implementation into this class as well. I'd be OK with making that a follow up ticket, though.

Reply:
Agreed 100%, but I'd definitely lean toward doing that as part of follow-up work.